Objects/longobject.c

```diff
@@ -5421,6 +5432,59 @@ int_bit_length_impl(PyObject *self)
     return NULL;
 }
 
+/*[clinic input]
+int.bit_count
+
+Number of ones in the binary representation of self.
+
+>>> bin(13)
+'0b1101'
+>>> (13).bit_count()
+3
+[clinic start generated code]*/
+
+static PyObject *
+int_bit_count_impl(PyObject *self)
+/*[clinic end generated code: output=2e571970daf1e5c3 input=a428900d3e39a606]*/
+{
+    Py_ssize_t ndigits, i, bit_count = 0;
+    PyLongObject *result, *x, *y;
```

Comment:
nitpick: since Python now uses C99, I prefer to declare variables where they are initialized. But I'm fine with keeping the code as it is.

Reply:
Done.